src/main/java/seedu/taassist/model/student/Student.java

```diff
@@ -85,6 +87,18 @@ public StudentModuleData findStudentModuleData(ModuleClass targetClass) {
         return moduleDataList.findElement(new StudentModuleData(targetClass));
```

Comment:
Add a note for this method that the student must be assigned to `targetClass`. May this can be checked with an `assert`.